ruby/example_code/lambda/lambda_function.rb

```diff
@@ -0,0 +1,32 @@
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0
+
+# frozen_string_literal: true
+
+# snippet-start:[ruby.example_code.lambda.handler.increment]
+require "logger"
+
+# A function that increments a whole number by one (1) and logs the result.
+# Requires a manually-provided runtime parameter, 'number', which must be Int
+#
+# @param event [Hash] contains parameters sent when the function is invoked
+# @param context [Hash] contains methods and properties that provide information
+# about the invocation, function, and execution environment.
+def lambda_handler(event:, context:)
+  logger = Logger.new($stdout)
+  log_level = ENV["LOG_LEVEL"]
+  logger.level = case log_level
+                 when "debug"
+                   Logger::DEBUG
+                 when "info"
+                   Logger::INFO
+                 else
+                   Logger::ERROR
+                 end
+  logger.debug("This is a debug log message.")
+  logger.info("This is an info log message. Code executed successfully!")
+  number = event["number"].to_i
+  incremented_number = number + 1
+  logger.info("You provided #{number.round} and it was incremented to #{incremented_number.round}")
```

Comment:
Does this also return the incremented number as well as log it? That's part of the intent of the example--to show retrieving parameters out of the payload returned from function invoke.

Reply:
I can add the return, but I'm honestly a little confused about the purpose of a function returning a value. Specifically, I'm accustomed to having a function perform a duty in isolation based on a trigger in a decoupled fashion, not necessarily having a secondary application waiting on it to return a value as part of a different set of logic. My opinionated side thinks this is falling into the exact pattern that Lambda is supposed to free us from, i.e. it seems sketchy to have a Ruby script calling Lambda which executes a Ruby script and waiting for a reply. Does that make sense?

Reply:
What you say makes sense, but I think there are scenarios where you might invoke a Lambda and wait for a result.
One that I've seen is a Lambda fronted by API Gateway to make a REST service backed by Lambda functions, and the paradigm in that case is to return the status code and body in a Python dict from the Lambda function (for Python Lambda).
Or you might have compute-intensive or resource-sensitive function that can invoked by an app or user who only has rights to invoke the function but not to access underlying services.